clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2017,7 +2041,13 @@ SYCLIntegrationHeader::SYCLIntegrationHeader(DiagnosticsEngine &_Diag,
 // -----------------------------------------------------------------------------
 
 bool Util::isSyclAccessorType(const QualType &Ty) {
-  return isSyclType(Ty, "accessor", true /*Tmpl*/);
+  if (const CXXRecordDecl *RecTy = Ty->getAsCXXRecordDecl())
```

Comment:
This seems odd here.  Why can't isSyclType handle this?

Reply:
ok